### PR TITLE
 Use Optional for nullable fields in SchemaTablePrefix 

### PR DESCRIPTION
**Pull request by kokosing**
Prefer SchemaTablePrefix#getSchema to create optional schema name

### CODE REVIEW

**Comment by kokosing** [comment]
Copy of https://github.com/prestodb/presto/pull/11086

File: presto-spi/src/main/java/io/prestosql/spi/connector/SchemaTablePrefix.java [hunk 1]
```diff
@@ -14,64 +14,80 @@
...
     public SchemaTablePrefix(String schemaName, String tableName)
     {
-        this.schemaName = checkNotEmpty(schemaName, "schemaName");
-        this.tableName = checkNotEmpty(tableName, "tableName");
+        this.schemaName = Optional.of(checkNotEmpty(schemaName, "schemaName"));
+        this.tableName = Optional.of(checkNotEmpty(tableName, "tableName"));
     }
 
+    /**
+     * @deprecated replaced by {@link SchemaTablePrefix#getSchema()}
```
Comment (electrum) [inline presto-spi/src/main/java/io/prestosql/spi/connector/SchemaTablePrefix.java]: Nit: the class name isn't need when the method is in the same class

File: presto-spi/src/main/java/io/prestosql/spi/connector/SchemaTablePrefix.java [hunk 2]
```diff
@@ -14,64 +14,80 @@
...
+    }
+
+    public Optional<String> getTable()
     {
         return tableName;
     }
 
     public boolean matches(SchemaTableName schemaTableName)
     {
-        // null schema name matches everything
```
Comment (electrum) [inline presto-spi/src/main/java/io/prestosql/spi/connector/SchemaTablePrefix.java]: Let's leave the comment but change to
```java
// empty prefix matches everything

Comment (kokosing) [reply presto-spi/src/main/java/io/prestosql/spi/connector/SchemaTablePrefix.java]: Isn't this already written one line below, but in java?

Anyway, added a comment.

**Pull request merged**